crates/shared/src/price_estimation/native/coingecko.rs

```diff
@@ -71,101 +72,167 @@ impl CoinGecko {
         })
     }
 
-    pub async fn send_request_price_in_eth(&self, token: Token) -> NativePriceEstimateResult {
+    async fn bulk_fetch_denominated_in_eth(
+        &self,
+        tokens: &[&Token],
+    ) -> Result<HashMap<Token, f64>, PriceEstimationError> {
         let mut url = crate::url::join(&self.base_url, &self.chain);
```

Comment:
We can have a size check here for `tokens` with an early return is exceed the bulk size.

Reply:
Initially thought about that but ultimately I think this is a configuration error and you will get an error anyway when sending the big request.